ml-agents/mlagents/trainers/tests/test_training_status.py

```diff
@@ -0,0 +1,45 @@
+import os
+import unittest
+import json
+
+from mlagents.trainers.training_status import (
+    StatusType,
+    StatusMetaData,
+    GlobalTrainingStatus,
+)
+
+
+def test_stats_reporter_store_restore(tmpdir):
+    trainingstatus = GlobalTrainingStatus("Category1")
+    path_dir = os.path.join(tmpdir, "test.json")
+
+    trainingstatus.store_parameter_state(StatusType.LESSON_NUM, 3)
+    GlobalTrainingStatus.save_state(path_dir)
+
+    with open(path_dir, "r") as fp:
+        test_json = json.load(fp)
+
+    assert "Category1" in test_json
+    assert StatusType.LESSON_NUM.value in test_json["Category1"]
+    assert test_json["Category1"][StatusType.LESSON_NUM.value] == 3
+    assert "metadata" in test_json
+
+    statsreporter_new = GlobalTrainingStatus("Category1")
+    GlobalTrainingStatus.load_state(path_dir)
+    restored_val = statsreporter_new.restore_parameter_state(StatusType.LESSON_NUM)
```

Comment:
Also test that restore_parameter_state() on an unknown category or StatusType returns None instead of raising an exception.

Reply:
Added test for these.